### DISCUSSION TITLE
Compiler Error

### DISCUSSION THREAD

**Discussion by FrankyBBillions** [marcel-licence/esp32_multitrack_looper#5] [General]
Hello Marcel,
And thank you for this project, I have completed the hardware assembly and having trouble compiling the code as below. I look forward to your guidance, thanks in advance.

<start>
Arduino: 1.8.13 (Windows 7), Board: "ESP32 Dev Module, Enabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 115200, Verbose"

1. status_module:197:51: error: embedded '\0' in format [-Werror=format-contains-nul]
     sprintf(statusMsg, "%s: %0.2f\0", descr, value);
2. status_module:206:48: error: embedded '\0' in format [-Werror=format-contains-nul]
     sprintf(statusMsg, "%s: %d\0", descr, value);
3. status_module:215:36: error: embedded '\0' in format [-Werror=format-contains-nul]
     sprintf(statusMsg, "%s\0", text);

exit status 1
embedded '\0' in format [-Werror=format-contains-nul]
<end>

Please excuse my ignorance if I have overlooked something, I’m an old guy but relatively new to coding. 

Take Care,
Frank

**Comment by marcel-licence** [comment]
Hello Frank,

thank you for your message. My idea was to ensure that there is a null at the end of the new created string because I was not sure if that was generated automatically by sprintf.
You can modify the code as follows:

    memset(statusMsg, 0, sizeof(statusMsg));
    sprintf(statusMsg, "%s: %0.2f", descr, value);

With memset the buffer will be reset and the \0 can be removed.

Hope this helps.

Best regards,
Marcel

**Reply by FrankyBBillions** [reply]
Thank you Marcel,

Perfect.

Sincere Regards,
Frank
